File: Try_GameEngine5/src/com/example/try_gameengine/framework/IGameController.java
package com.example.try_gameengine.framework;

import android.graphics.Canvas;
import android.view.MotionEvent;
import android.view.SurfaceHolder;

public interface IGameController {
	public void start();
	public void stop();
	public void showWin();
	public void showLose();
	public void onTouchEvent(MotionEvent event);
	public void setSurfaceHolder(SurfaceHolder surfaceHolder);
	public void runStart();
	public void surfaceChanged(SurfaceHolder holder, int format, int width,
			int height);
}

File: Try_GameEngine5/src/com/example/try_gameengine/framework/IGameController.kt
package com.example.try_gameengine.framework

import android.view.MotionEvent
import android.view.SurfaceHolder

interface IGameController {
    fun start()
    fun stop()
    fun showWin()
    fun showLose()
    fun onTouchEvent(event: MotionEvent?)
    fun setSurfaceHolder(surfaceHolder: SurfaceHolder?)
    fun runStart()
    fun surfaceChanged(
        holder: SurfaceHolder?, format: Int, width: Int,
        height: Int
    )

    fun setFlag(flag: Int)
}
